lib/puppet/functions/katello/build_dn.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'yaml'
+# @summary
+#   Convert an array of attribute pairs to a DN string, ignoring empty values
+#
+# @example Pass a set of
+#   $client_dn = katello::build_dn([['CN', 'foo.example.com'], ['O', 'my_org']])
+Puppet::Functions.create_function(:'katello::build_dn') do
+  # @param options
+  #
+  # @return String
+  dispatch :build_dn do
+    param 'Array', :options
```

Comment:
I think this is a stricter validation of what you describe:
```suggestion
    param 'Array[Tuple[String[1], Optional[String[1]]]', :options
```